ms/MeasurementSets/MeasurementSet.cc

```diff
@@ -1075,6 +1090,45 @@ Record MeasurementSet::msseltoindex(const String& spw, const String& field,
   return retval;
 }
 
+void MeasurementSet::_checkInternalConsistency() const {
+    Int nrows = 0;
+    String colname, tablename;
+    for (uInt i=0; i<4; ++i) {
+        switch (i) {
+        case 0:
+            nrows = antenna_p.nrow();
+            tablename = antenna_p.tableName();
+            colname = columnName(MSMainEnums::ANTENNA1);
+            break;
+        case 1:
+            nrows = antenna_p.nrow();
+            tablename = antenna_p.tableName();
+            colname = columnName(MSMainEnums::ANTENNA2);
+            break;
+        case 2:
+            nrows = dataDesc_p.nrow();
+            tablename = dataDesc_p.tableName();
+            colname = columnName(MSMainEnums::DATA_DESC_ID);
+            break;
+        case 3:
+            nrows = field_p.nrow();
+            tablename = field_p.tableName();
+            colname = columnName(MSMainEnums::FIELD_ID);
+            break;
+        default:
+            break;
+        }
+        Int mymax = max(ROScalarColumn<Int>(*this, colname).getColumn());
+        ThrowIf(
+            mymax >= nrows,
+            "Illegal " + colname + " value " + String::toString(mymax)
+            + " found in main table. " + dataDesc_p.tableName() + " only has "
```

Comment:
this always prints the data description table row, though nrow refers to another table, typo?